src/wakefs/fuse.cpp

```diff
@@ -217,16 +224,56 @@ bool run_in_fuse(fuse_args &args, int &status, std::string &result_json) {
   (void)close(STDOUT_FILENO);
   (void)close(STDERR_FILENO);
 
-  do waitpid(pid, &status, 0);
-  while (WIFSTOPPED(status));
+  bool has_timeout = args.command_timeout > 0;
+  pid_t timeout_pid = -1;
+
+  // Launch timer process
+  if (has_timeout) {
+    timeout_pid = fork();
+    if (timeout_pid < 0) {
+      std::cerr << "wakebox: failed to fork timout process" << std::endl;
+      exit(1);
+    }
+
+    if (timeout_pid == 0) {
+      sleep(args.command_timeout);
+      exit(124);
+    }
+  }
+
+  pid_t wait_pid;
+  while ((wait_pid = wait(&status)) != -1) {
+    if (wait_pid == timeout_pid && WIFEXITED(status)) {
+      kill(payload_pid, SIGKILL);
+
+      struct timeval stop;
+      gettimeofday(&stop, 0);
+      std::string output;
+      args.daemon.disconnect(output);
+      RUsage usage = {};
+      return collect_result_metadata(output, start, stop, payload_pid, 124, usage, true,
```

Comment:
It doesn't really matter what the specific number is, as anything that care about knowing the real reason can check the result json

Reply:
Actually, I think we should return zero when isolate-retcode is set, as wakebox itself didn't fail, just the payload

Reply:
I think pretending its 124 is somewhat helpful for debugging potentially. isolate-retcode is just meant to make wakebox return 0 as long as it doesn't fail right? I think this number is the return code of the payload isn't it?

Reply:
oh my bad, this is the path that threads through into the result json, rather than the exterior